index.js

```diff
@@ -336,7 +344,7 @@ function session(options) {
         });
 
         return writetop();
-      } else if (storeImplementsTouch && shouldTouch(req)) {
+      } else if ((alwaysTouchUnmodified || rollingSessions) && storeImplementsTouch && shouldTouch(req)) {
```

Comment:
Should this extra logic be inside the `shouldTouch` function? Or if not, why not? I would expect that function is all that would be called to determine if a touch occurs, and this is controlling the touching.